.github/actions/setup/action.yml

```diff
@@ -20,17 +20,30 @@ runs:
       with:
         version: nightly
 
+    # Workaround for parallel building with forge:
+    # https://github.com/foundry-rs/foundry/issues/4736
+    - name: Install solc
+      shell: bash
+      run: cargo install svm-rs && svm install 0.8.13
+
     - name: Setup protoc
       uses: arduino/setup-protoc@v1
 
     - name: Install node dependencies
       shell: bash
       run: pnpm install --frozen-lockfile
 
+    - name: Cache turbo build
+      uses: actions/cache@v3
+      with:
+        path: .turbo
+        key: ${{ runner.os }}-turbo-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-turbo-
```

Comment:
pulled this from https://github.com/vercel/turbo/pull/2761

but `runner.os` being the only unique key for the cache seems a little aggressive? is turbo smart enough to figure out what cached things to use across all our builds/commits?

Reply:
I [forked austinwoon's demo](https://github.com/Handfish/turbo-repo-cache-with-github-actions), took a look at his commits and recreated his build examples so I could view them. I think turbo is smart enough to cache properly with that restore-key. It acts as if it would on your local machine.

Reply:
Branch name might be useful though because in the case two people are committing at the same time they will be fighting over the cache. According to the docs, multiple restore-keys can be provided and have a cascading fallback, so maybe it'd be good to take advantage of that.